libs/blocks/event-rich-results/event-rich-results.js

```diff
@@ -0,0 +1,78 @@
+import { createTag } from '../../utils/utils.js';
+
+function buildData(el) {
+  const data = new Map();
+  const rows = el.children;
+  [...rows].forEach((row) => {
+    const key = [...row.children][0]?.innerText?.toLowerCase();
+    const value = [...row.children][1]?.innerText;
+    data.set(key, value);
+  });
+  return data;
+}
+
+function getLocation(data) {
+  if (data.get('location-type') === 'Place') {
+    return {
+      '@type': 'Place',
+      name: data.get('location-name'),
+      address: {
+        '@type': 'PostalAddress',
+        streetAddress: data.get('location-address-street'),
+        addressLocality: data.get('location-address-locality'),
+        postalCode: data.get('location-address-postal-code'),
+        addressRegion: data.get('location-address-region'),
+        addressCountry: data.get('location-address-country'),
```

Comment:
I know for metadata we only accept the `-` between keys like `gnav-source` and not a space. This being a block is it a concern that authors have to add `-` in place of spaces in the keys or the schema fails? Anyone have thoughts?

Otherwise looks good. 
Same [result URL](https://search.google.com/test/rich-results/result/r%2Fevents?id=IDy7h4Oh3HCEucXZ3JoOHw)

Reply:
@rgclayton I adapted the code at https://github.com/adobecom/milo/pull/511/commits/5d5cda6cb3209c6ec35a78006c27c5c4a534b7cf to allow authors to enter either whitespace or '-' in the property name.